### PR TITLE
Add ocm support

### PR DESCRIPTION
**Pull request by anishasthana**

### CODE REVIEW

File: controllers/appwrapper_controller.go [hunk 1]
```diff
@@ -90,9 +89,17 @@ var kubeClient *kubernetes.Clientset
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
 func (r *AppWrapperReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+
+	if ocmClusterID == "" && useMachineSets == false {
+		err := getAccessTokens(r)
+		if err != nil {
+			useMachineSets = true
```
Comment (asm582) [inline controllers/appwrapper_controller.go:96]: we should probably log the error. I think users will use either machineset or machinepool.

File: controllers/machinepools.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
+package controllers
```
Comment (asm582) [inline controllers/machinepools.go:1]: Very nice, thank you for creating a separate file :)

File: main.go [hunk 3]
```diff
@@ -43,9 +44,11 @@ var (
 
 func init() {
 	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(configv1.Install(scheme))
```
Comment (asm582) [inline main.go:47]: curious, would like to understand better changes done to init function

**Review by asm582 — COMMENTED** [review]
Thank you for the PR, please resolve minor changes and I will approve

File: main.go (hunk 3 above)
Comment (anishasthana) [reply main.go:47]: So, unless we do this the controller is unnable to read the configv1 object type. In general you end up needing to do something like this for any new CRD you want your controller to interact with.

**Review by asm582 — APPROVED** [review]
LGTM 🎉

**Comment by asm582** [comment]
@Maxusmusti I have tested with both modes, can you please review and approve?

**Review by Maxusmusti — APPROVED** [review]
LGTM

**Pull request merged**